### PR TITLE
move sample to example

### LINKED ISSUE #89: Rename Sample target name
Move current code in CMake Sample target to `test` and make the default project empty

### PR DESCRIPTION
**Pull request by NOOBDY**
change the original executable to a library and move the sample code to `/example` folder

- part of #156 
- resolve #89

### CODE REVIEW

**Review by ntut-rick — COMMENTED** [review]
IWOMM

File: CMakeLists.txt [hunk 1]
```diff
@@ -87,15 +84,22 @@ set(INCLUDE_FILES
...
+    ${EXAMPLE_DIR}/src/Giraffe.cpp
+    ${EXAMPLE_DIR}/src/GiraffeText.cpp
+    ${EXAMPLE_DIR}/src/Cat.cpp
+    ${EXAMPLE_DIR}/src/main.cpp
 
-    ${INCLUDE_DIR}/App.hpp
-    ${INCLUDE_DIR}/Giraffe.hpp
-    ${INCLUDE_DIR}/GiraffeText.hpp
-    ${INCLUDE_DIR}/Cat.hpp
-    ${INCLUDE_DIR}/config.hpp
```
[the remark this answers is not in the document]
Comment (Dragonfly911117) [reply CMakeLists.txt:98]: It's at line 59.

**Review by Dragonfly911117 — COMMENTED** [review]
So I was thinkg if we should separate `Tests` and `Example` related CMake configuations to each `CMakeLists.txt` in their directories and have `${REPO_ROOT}/CMakeLists.txt` add them.

**Comment by NOOBDY** [comment]
I don't think it needs to be that complicated

File: CMakeLists.txt [hunk 2]
```diff
@@ -104,67 +109,75 @@ set(TEST_FILES
...
+
 enable_testing()
 
-add_executable(Tests
+add_executable(Tests EXCLUDE_FROM_ALL
     ${TEST_FILES}
     ${SRC_FILES}
 )
 target_link_libraries(Tests
-    GTest::gtest_main
```
Comment (Dragonfly911117) [inline CMakeLists.txt:137]: I think you forgot to put this back.
```suggestion
target_link_libraries(Tests
    GTest::gtest_main
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -104,67 +109,75 @@ set(TEST_FILES
...
 
-add_executable(AudioTest
+add_executable(AudioTest EXCLUDE_FROM_ALL
     ${TEST_DIR}/Interactive/Audio.cpp
     ${SRC_FILES}
 )
 target_link_libraries(AudioTest
-    GTest::gtest_main
-    ${DEPENDENCY_LINK_LIBRARIES}
+    PTSD
```
Comment (Dragonfly911117) [inline CMakeLists.txt:173]:
```suggestion
target_link_libraries(AudioTest
    PTSD
    GTest::gtest_main
```

**Review by Dragonfly911117 — CHANGES_REQUESTED** [review]

**Review by ntut-rick — APPROVED** [review]

**Review by Dragonfly911117 — APPROVED** [review]

**Pull request merged**